hackathon_site/event/test_api.py

```diff
@@ -52,42 +52,77 @@ def test_user_has_profile(self):
         self.assertEqual(response.json(), serializer.data)
 
 
-class CurrentTeamTestCase(SetupUserMixin, APITestCase):
+class JoinTeamTestCase(SetupUserMixin, APITestCase):
     def setUp(self):
         super().setUp()
         self.group = Group.objects.create(name="Test Users")
         self.user.groups.add(self.group)
         self.team = Team.objects.create()
 
         self.profile = Profile.objects.create(user=self.user, team=self.team)
+        self.team_code = self.team.team_code
+        self.view_name = "api:event:join-team"
 
-        self.view = reverse("api:event:current-team")
+    def _build_view(self, team_code):
+        return reverse(self.view_name, kwargs={"team_code": team_code})
+
+    def test_invalid_key(self):
+        self._login()
+        response = self.client.post(self._build_view("56ABD"))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_join_full_team(self):
+        self._login()
+        team = self._make_full_event_team(self_users=False)
+        print(team.team_code)
+        response = self.client.post(self._build_view(team.team_code))
+        self.assertEqual(str(response.content), 'b\'{"detail":"Team is full"}\'')
 
     def test_user_not_logged_in(self):
-        response = self.client.get(self.view)
+        response = self.client.post(self._build_view("56ABD"))
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
 
     def test_user_has_no_profile(self):
-        """
-        When the user attempts to access the team, while it has no profile.
-        The user must be accepted or waitlisted to have formed a team.
-        """
         self.profile.delete()
         self._login()
-        response = self.client.get(self.view)
-        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        response = self.client.post(self._build_view("56ABD"))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
-    def test_user_has_profile(self):
+    def check_cannot_join(self):
+        old_team = self.profile.team
+        response = self.client.post(self._build_view(self.team.team_code))
+        self.user.refresh_from_db()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(old_team.pk, self.user.profile.team.pk)
+        self.assertEqual(Team.objects.filter(team_code=old_team.team_code).count(), 1)
+
+    def test_leave_with_order(self):
```

Comment:
this isn't the leave team view

Reply:
changed up the hardware consideration as per above comments.